Improve the API documentation, Module MD, and Package MD (#179)
ballerina-platform/module-ballerina-io: io-ballerina/Module.md

> ## Module Overview

Reviewer: ```suggestion
## Overview
```

This module provides file read/write APIs and console print/read APIs. The file APIs allow read and write operations on different kinds of file types such as bytes, text, CSV, JSON, and XML. Further, these file APIs can be categorized as streaming and non-streaming APIs.

The following diagram depicts the overview architecture of the I/O module.

![architecture](./resources/architecture.svg)

> The file I/O operations can be further categorized based on serialization and deserialization types such as:

Reviewer 2: ```suggestion
The file I/O operations can be categorized further based on the serialization and deserialization types such as:
```

- Bytes I/O
- Strings I/O
- CSV I/O
- JSON I/O
- XML I/O

### Console I/O

> The console I/O APIs, which help the user to read from the console as well as write to the console are as follows.

Reviewer 2: ```suggestion
The console I/O APIs, which help you to read from the console as well as write to the console are as follows.
```

> - print
> - println
> - readln

Reviewer: ```suggestion
- `io:print`
- `io:println`
- `io:readln`
```

### Bytes I/O
The bytes I/O APIs provide the reading and writing APIs in both streaming and non-streaming ways. Those APIs are,
- `io:fileReadBytes`
- `io:fileReadBlocksAsStream`
- `io:fileWriteBytes`
- `io:fileWriteBlocksFromStream`

### Strings I/O
The strings I/O APIs provide the reading and writing APIs in 3 different ways:

> 1. Read the whole file content as a string and write a given string to a file

Reviewer 2: ```suggestion
1. Read the complete file content as a string and write a given string to a file
```

> 2. Read the whole file content as a set of lines and write a given set of lines to a file

Reviewer 2: ```suggestion
2. Read the complete file content as a set of lines and write a given set of lines to a file
```

> 1. Read the whole file content as a set of lines and write a given set of lines to a file
> 1. Read the whole file content as a stream of lines and write a given stream of lines to a file

Reviewer: ```suggestion
2. Read the whole file content as a set of lines and write a given set of lines to a file
3. Read the whole file content as a stream of lines and write a given stream of lines to a file
```

> 3. Read the whole file content as a stream of lines and write a given stream of lines to a file

Reviewer 2: ```suggestion
3. Read the complete file content as a stream of lines and write a given stream of lines to a file
```

## Module Overview

> This module provides file read/write APIs and console print/read APIs. The file APIs allow read and write operations on different kinds of file types such as bytes, text, CSV, JSON, and XML. Further, these file APIs can be categorized as streaming and non-streaming APIs.

Reviewer 3: It seems there is a space at the start. Is that expected ?

Author: Yes, it is intentional.

> This module provides two major types of APIs: file read/write APIs and console print/read APIs. The file APIs allow read and write operations on different kinds of file types such as bytes, text, CSV, JSON, and XML. Further, these file APIs can be categorized as streaming and non-streaming APIs.

Reviewer: ```suggestion
This module provides file read/write APIs and console print/read APIs. The file APIs allow read and write operations on different kinds of file types such as bytes, text, CSV, JSON, and XML. Further, these file APIs can be categorized as streaming and non-streaming APIs.
```

The following diagram depicts the overview architecture of the I/O module.

![architecture](./resources/architecture.svg)

The file I/O operations can be further categorized based on serialization and deserialization types such as:
- Bytes I/O
- Strings I/O
- CSV I/O
- JSON I/O
- XML I/O

> 

Reviewer 3: I think we can remove this additional new line.

Author: Fixed in here https://github.com/ballerina-platform/module-ballerina-io/pull/179/commits/fd0da0ae53df77cfef77ca08159e7a84a0ea28b6
